README.md
# DC-Plugins

<p>
<img src="https://img.shields.io/badge/license-Apache%202-blue"/>
<img src="https://img.shields.io/npm/v/@dvgis/dc-plugins?logo=npm&color=orange" />
<img src="https://img.shields.io/npm/dm/@dvgis/dc-plugins?logo=npm"/>
</p>

[**🇨🇳 中文**](./README_zh.md) | [**🇬🇧English**](./README.md)

> DC-SDK plug-in library, the plug-in library includes animation, effects, textures, heatmap, clusterLayer, and Mapv.

## Home

> http://dc.dvgis.cn

## Installation

`CDN`

```html
<!--Plugins Package-->
<script src="libs/dc-sdk/plugins/dc.plugins.min.js"></script>
```

`NPM / YARN`

```shell
yarn add @dvgis/dc-plugins
npm install  @dvgis/dc-plugins
```

```js
import DcPlugins from  'dvgis/dc-plugins/dist/dc.plugins.min' //Plugins Package
```

## Start

```js
DC.use(DcPlugins)
```

## Documentation

[DC Plugins Api](https://resource.dvgis.cn/dc-api/dc-plugins/)

## Demo

| ![picture](http://dc.dvgis.cn/examples/images/layer/cluster_clustering.gif)  | ![picture](http://dc.dvgis.cn/examples/images/overlay/polyline_image_trail.gif) | ![picture](http://dc.dvgis.cn/examples/images/overlay/polyline_flow.gif) | ![picture](http://dc.dvgis.cn/examples/images/overlay/wall_trail.gif) |
| :---------------------------------------------------------------: | :-----------------------------------------------------------------------------: | :---------------------------------------------------------------------: | :-------------------------------------------------------------------: |
| ![picture](http://dc.dvgis.cn/examples/images/scene/start_animation.gif) | ![picture](http://dc.dvgis.cn/examples/images/scene/around_point.gif)  | ![picture](http://dc.dvgis.cn/examples/images/scene/circle_scan.gif?v=1) | ![picture](http://dc.dvgis.cn/examples/images/scene/radar_scan.gif) |
| ![picture](http://dc.dvgis.cn/examples/images/scene/snow.gif) | ![picture](http://dc.dvgis.cn/examples/images/scene/fog.png)  | ![picture](http://dc.dvgis.cn/examples/images/scene/brightness.png?v=2) | ![picture](http://dc.dvgis.cn/examples/images/scene/roaming_tracked.gif) |


[More>>](http://dc.dvgis.cn/#/examples)

## Copyright statement

```warning
1. The framework is a basic platform, completely open source, which can be modified and reconstructed by any individual or institution without our authorization.
2. We are not responsible for any problems arising from the modification of the framework by individuals and organizations.
3. Some industrial plug-ins and tools will be added in the later stage, and the code will be open source appropriately.
4. The package released by us may be used permanently and free of charge by any person or organization subject to:
  1) complete package reference;
  2) reserve this copyright information in the console output
We reserve the right of final interpretation of this copyright information.
```

## Thanks


CHANGES.md
# Change Log

### 1.12.0 - 2021-03-06

> 1. 添加圆形覆盖物的材质和对应的材质属性
> 2. 添加雷达材质和对应的材质属性
> 3. 添加墙体材质和对应的材质属性
> 4. 添加球体材质和对应的材质属性

### 1.11.1 - 2021-02-28

> 1. 修改风向图层无法隐藏的问题
> 2. 修改材质的目录结构
> 3. 添加圆形覆盖物的材质
> 4. 添加线条覆盖物的材质

### 1.11.0 - 2021-02-28

> 1. 完善扫描圈和扫描雷达的功能
> 2. 完善框架目录结构

### 1.10.3 - 2021-01-30

> 1. 修复天气效果初始化的问题
> 2. 将雾天的 fogColor 属性改为 color
> 3. 添加 Cesium 内置后期场景的设置

### 1.10.2 - 2021-01-23

> 1. 添加风向图层

### 1.10.1 - 2021-01-16

> 1. 优化漫游组件功能
> 2. 移除CzmlLayer 和 KmlLayer
> 3. 添加定点巡航驻留时间
> 4. 完善一些覆盖物纹理

### 1.10.0 - 2021-01-10

> 1. 添加一些覆盖物纹理
> 2. 完善轨迹线的纹理
> 3. 添加定点巡航动画

### 1.9.3 - 2021-01-04

> 1. 完善部分材质功能
> 2. 添加扫描线材质

### 1.9.2 - 2020-12-26

> 1. 完善漫游功能，修改为每一个path单独的时间控制

### 1.9.1 - 2020-12-20

> 1. 重构场景动画功能，支持手动暂停和播放

### 1.9.0 - 2020-12-05

> 1. 重构场景效果
> 2. 重组目录结构
> 3. 修复部分动画材质速度设置无效的问题

### 1.8.0 - 2020-11-30

> 1. 完善 webpack 配置
> 2. 添加天气模块

### 1.7.3 - 2020-11-23

> 1. 完善波纹圆shader

### 1.7.2 - 2020-10-31

> 1. 完善圆逐渐消逝的shader
> 2. 完善场景动画功能

### 1.7.1 - 2020-10-28

> 1. 添加聚合图层的样式设置

### 1.7.0 - 2020-9-19

> 1. 完善雨天和雪天的速度控制
> 2. 修改框架包node下导入的方式
> 3. 完善漫游功能

### 1.6.1 - 2020-8-15

> 1. 优化雾天效果代码
> 2. 优化场景亮度后期效果代码
> 3. 添加夜视场景功能

### 1.6.0 - 2020-8-12

> 1. 添加场景亮度后期效果
> 2. 优化webpack配置

### 1.5.0 - 2020-8-6

> 1. 修改场景动画监听事件的类型
> 2. 添加漫游控件

### 1.4.3 - 2020-7-31

> 1. 修改雾效果参数
> 2. 完善框架导入方式

### 1.4.2 - 2020-7-29

> 1. 移除 Mapv 相关功能
> 2. 添加代码混淆功能

### 1.4.0 - 2020-7-18

> 1. 修改流动材质属性
> 2. 添加图片流动材质

### 1.3.0 - 2020-6-25

> 1. 修改场景动画的实现
> 2. 修改材质属性设定
> 3. 添加流动墙体材质

### 1.2.0 - 2020-6-4

> 1. 修改聚合图层 billboard 地形遮挡问题
> 2. 修改热区图层无法设置点位问题

### 1.1.1 - 2020-5-21

> 1. 修改绕点飞行和相机飞行的无法使用的问题
> 2. 图层类型获取错误

### 1.0.0 - 2020-5-12

> 1. 初始化


webpack.conf.js
/*
 * @Author: Caven
 * @Date: 2020-01-18 18:22:23
 * @Last Modified by: Caven
 * @Last Modified time: 2020-05-12 13:03:42
 */

const path = require('path')
const webpack = require('webpack')
const MiniCssExtractPlugin = require('mini-css-extract-plugin')
const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin')

function resolve(dir) {
  return path.join(__dirname, '.', dir)
}

module.exports = env => {
  const IS_PROD = (env && env.production) || false
  const publicPath = IS_PROD ? '/' : '/'
  let plugins = [
    new MiniCssExtractPlugin({
      filename: IS_PROD ? '[name].min.css' : '[name].css',
      allChunks: true
    })
  ]
  if (IS_PROD) {
    plugins.push(new OptimizeCssAssetsPlugin())
    plugins.push(new webpack.NoEmitOnErrorsPlugin())
  }
  return {
    entry: {
      'dc.plugins': ['entry']
    },
    devtool: IS_PROD ? false : 'cheap-module-eval-source-map',
    output: {
      filename: IS_PROD ? '[name].min.js' : '[name].js',
      path: path.resolve(__dirname, 'dist'),
      publicPath: publicPath,
      sourcePrefix: ''
    },
    amd: {
      toUrlUndefinded: true
    },
    node: {
      fs: 'empty'
    },
    module: {
      unknownContextCritical: false,
      rules: [
        {
          test: /\.js$/,
          exclude: /node_modules/,
          loader: 'babel-loader',
          query: {
            presets: ['@babel/preset-env'],
            compact: false,
            ignore: ['checkTree']
          }
        },
        {
          test: /\.css$/,
          use: [
            MiniCssExtractPlugin.loader,
            {
              loader: 'css-loader'
            },
            {
              loader: 'sass-loader'
            }
          ]
        },
        {
          test: /\.scss$/,
          use: [
            MiniCssExtractPlugin.loader,
            {
              loader: 'css-loader'
            },
            {
              loader: 'sass-loader'
            }
          ]
        },
        {
          test: /\.(png|jpg|gif|svg|eot|ttf|woff|woff2)$/,
          loader: 'url-loader',
          options: {
            limit: 20000
          }
        },
        {
          test: /\.glsl$/,
          loader: 'webpack-glsl-loader'
        }
      ]
    },
    resolve: {
      extensions: ['.js', '.json', '.css'],
      alias: {
        '@': resolve('src'),
        entry: './src/index.js'
      }
    },
    plugins
  }
}
